WIP: Support Windows

jakirkham opened the pull request:
Adds some WIP changes to the feedstock/recipe to try and build this on Windows using MSYS2. Currently doesn't work, but should be a good starting point for anyone interested in pursuing this.

jakirkham committed:
Drop the `skip` on Windows

Go ahead and enable Windows build of `fswatch`. Currently there are no
build scripts to build on Windows. However this is done in preparation
of the recipe/feedstock for building on Windows.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.5

Now that the `skip` on Windows has been dropped, re-render to configure
the feedstock to enable Windows builds.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Add automake version to custom feedstock matrix

As Windows will need to rerun `autoconf` and `automake` and the
`automake` version must be specified with the package, try adding the
version of `automake` to use to the custom feedstock matrix.

jakirkham committed:
Skip builds based on `AM_VERSION`'s value

Only build on Unix when `AM_VERSION` is an empty string. Only build on
Windows when `AM_VERSION` is set. This results in a reduction of the
custom feedstock matrix such that Windows builds include the
`AM_VERSION` environment variable and Unix builds do not.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.5

Now that the `AM_VERSION` variable is set for Windows builds, re-render
to add this variable to the AppVeyor configuration file.

jakirkham commented:
Ran into issue ( https://github.com/emcrisostomo/fswatch/issues/121 ). AFAICT this insurmountable without changes upstream.

jakirkham committed:
Build on Windows with MSYS2

Uses the MinGW-w64 toolchain to build `fswatch` on Windows. Currently
this is the only supported way to build on Windows. Reuses the Unix
build scripts with some tweaking to accomplish this.

jakirkham commented:
Closing as stale
